common/cards/card-plugins/effects/chainmail-armor.js

```diff
@@ -0,0 +1,50 @@
+import {GameModel} from '../../../../server/models/game-model'
+import EffectCard from './_effect-card'
+
+/**
+ * @typedef {import('common/types/cards').CardPos} CardPos
+ */
+
+class ChainmailArmorEffectCard extends EffectCard {
+	constructor() {
+		super({
+			id: 'chainmail_armor',
+			name: 'Chainmail Armor',
+			rarity: 'common',
+			description: 'Prevents damage from all effect cards.',
+		})
+	}
+
+	/**
+	 * @param {GameModel} game
+	 * @param {string} instance
+	 * @param {CardPos} pos
+	 */
+	onAttach(game, instance, pos) {
+		const {otherPlayer} = pos
+
+		otherPlayer.hooks.onAttack[instance] = (attack, pickedSlots) => {
+			if (attack.target.index !== pos.rowIndex || attack.type === 'ailment')
```

Comment:
You don't need to check for ailment if you've got a check right after filtering out anything that isn't an effect attack